Fix languages order in set TTS language dialog

lukstbit opened the pull request:
## Purpose / Description

The changes in this PR order alphabetically the languages in "Set TTS language" dialog. Also refactors and improves the code a bit.

## Fixes

Fixes #12345

## How Has This Been Tested?

Ran the tests, ran the app.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

david-allison reviewed (approved):
Could you add a `@NeedsTest`

**Implementer's choice**

 `compareBy` may be easier to read. 

There's a few other potential solution rather than using `.add()` in a `forEach`: using the `addAll` or `+`overridden method will save a few wasteful allocations, but it's a small dialog and I'd go for readability over performance here

```kotlin
availableTtsLocales.sortedWith(compareBy { it.displayName })
```

lukstbit commented:
Added _@NeedsTest("ensure languages are sorted alphabetically in the dialog")_.
Also implemented the suggestions, went with `addAll` as I think it reads better the combination add() then addAll.
